pkg/controllers/pod_controller.go

```diff
@@ -102,9 +113,187 @@ func SetupPodIndexes(ctx context.Context, indexer client.FieldIndexer) error {
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
 func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// Get Pod from apiserver.
```

Comment:
We should only have reconcile invoked for pods created via jobset and use exclusive placement. We should have a filter for that when setting up the reconciler with the manager, pls check how to do that in kubebuilder.

Reply:
This was already done in the previous PR, so it doesn't show up in the diff here, see the `SetupWithManager` function.